api/agent/protocol/http.go

```diff
@@ -62,12 +63,18 @@ func (h *HTTPProtocol) Dispatch(ctx context.Context, ci CallInfo, w io.Writer) e
 	span, _ = opentracing.StartSpanFromContext(ctx, "dispatch_http_write_response")
 	defer span.Finish()
 
-	buf := bufPool.Get().(*bytes.Buffer)
+	buf := BufPool.Get().(*bytes.Buffer)
 	buf.Reset()
-	defer bufPool.Put(buf)
+	defer BufPool.Put(buf)
+
+	reader := NewClampReader(resp.Body)
```

Comment:
i'm worried about doubly imposing this limit with https://github.com/fnproject/fn/blob/master/api/server/server_options.go#L20 -- that guy has the benefit of limiting the request body for every handler, but plumbing wise checking for that error from in here sucks. some options I guess. at least, there's some dupe code for limit reader stuff it looks like.

Reply:
Doesn't that pertain to gin contexts (specifically request.Body) though? This is the response, and it is a brand new http response we synthesize here.